### ISSUE TITLE
Open log icon button lacks text alternative and has incorrect role

### ISSUE THREAD

**Issue by henrycatalinismith** [zetkin/call.zetk.in#268]
## Steps to reproduce

1. Go to the call page.
2. Press `⌘ + F5` to start VoiceOver.
3. Focus the open log button.

## Expected outcome

It should be announced as a button element with a text label explaining its purpose. See WCAG success criteria [1.1.1](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html) and [4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) for more information.

## Actual outcome

It's announced as a link and has no label.

![VoiceOver's focus indicator on the button and the VoiceOver announcement transcript saying simply "Link"](https://user-images.githubusercontent.com/566159/191568104-5ac794e5-6f5f-4b60-accc-930c4ec99f39.png)

## Severity

The success criterion covering this issue is WCAG level A – the minimum level. Most organizations aim for a conformance level of AA, which includes all level A criteria. This is a severe issue as access to the modal functionality that the button opens would require a user to randomly click things, explore the resulting pages, and then memorize the location of the button once identified.

## Recommendation

* Use `<button>` rather than `<a>` for this element.
* Add an `aria-label` such as `Log`, `Open log`, `Call log`, or `Open call log etc.
